### DISCUSSION TITLE
Question regarding SAMD PWM

### DISCUSSION THREAD

**Discussion by Bexin3** [khoih-prog/ESP_WiFiManager#109] [Q&A]
Hello, first of all id like to apologise for cross posting, however that part is now closed. I tried to use my own code you can find on my profile on seeding zero, and had no luck, even though Arduino zero works with it perfectly, any idea what am I doing wrong?

**Comment by khoih-prog** [comment]
HI @Bexin3 

As you didn't use the [**SAMD_PWM**](https://github.com/khoih-prog/SAMD_PWM) library in [**your code**](https://github.com/Bexin3/Fast-Analog-to-PWM-for-SAMD21/blob/main/Analog_to_PWM.ino), I suggest you post the question on **Arduino Forum**, where there are many SAMD experts there can help you out.

You also have to post the [**MRE**](https://stackoverflow.com/help/minimal-reproducible-example) as well as what the problems you got. I think no one can spend time to download the code, try debugging for you, w/o knowing what's to search for and fix.

Good Luck,

PS:

1. The compile is OK for both Arduino Zero and Seeed Zero

![Selection_258](https://user-images.githubusercontent.com/57012152/212126550-7b261daf-61d9-4885-9c74-6042c7022a3f.png)

2. Next time, please don't cross post like this. The correct place would be **Arduino Forum** or [**Seeedstudio Forum**](https://forum.seeedstudio.com/)

**Reply by sgeklorer** [reply]
Hi khoih-prog,

It is actually really difficult that you archived the SAMD timer library project: https://github.com/khoih-prog/SAMD_TimerInterrupt

As a result of doing so, there is now no way to ask questions on that forum or to work on the code. Since you haven't listed any contact details (e.g. email) there is no way to reach you. The library is still available via the Arduino IDE so there is going to be a constant stream of users (of various abilities) coming to the GitHub project page and wondering what's going on.

Can you at least post a note on that project telling users why the project has been archived and what will happen in the future?

I am considering using the library in a project, but will have to reconsider if the project is "dead" or has unfixable problems: why else close it?

I've been through the code quite thoroughly over the past couple of days, and it's all really good stuff, it would just benefit from a very rigorous edit (lots of copy-pasted comments from other files that don't directly apply to the new file, for example). But again, if the project is dead then what's the point?

Happy to email you privately about this, but again, there's no contact information on your profile and GitHub stupidly removed the private messaging feature.

Thanks again for contributing the timer project, it's really neat,

Sgeklor

**Reply by Bexin3** [reply]
Hi sgeklorer, I am not the original developer so cant answer all of your questions, but I went quite extensively over this code today, and even though there are small features you may want to add I didn't find any bugs. Only issue is if you want to change duty cycle I recommend doing it directly as it takes too much time using the function. In my last project my core code was kinda inspired from here, though I have included everything in a file instead of making a library, and removed some features as well as adding some.

**Comment by khoih-prog** [comment]
Please have a look at [**Query: some dependencies (eg ESP_DoubleResetDetector) have been archived- safe to keep using?** #132](https://github.com/khoih-prog/ESPAsync_WiFiManager/discussions/132#discussioncomment-4651409)

> **Archived** library only means that library is **read-only** now. The reason is either
> 
> - it's quite **stable**
> - not intended to be updated
> - **not popular** and not intended to spend more time
> ...
> 
> But don't worry, those libraries will continue to be maintained / updated as needed. For example the **DRD**, etc.
> 
> The question / issue can be posted on **Arduino Forum**, etc. so that many other experts there can help and I'm aware of it to fix bug or update as **needed.**

I'll certainly will unarchive any library if there are some requests to use it. For example these will be unarchived now

- [**SAMD_PWM**](https://github.com/khoih-prog/SAMD_PWM)
- [**SAMD_TimerInterrupt**](https://github.com/khoih-prog/SAMD_TimerInterrupt)
